### ISSUE TITLE
"storeconfig" creates unecessary "acme/storageFile" entry when not needed

### ISSUE THREAD

**Issue by bitsofinfo** [containous/traefik#2547]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?
 
1. **Ran `storeconfig` for the following TOML against consul**

This works fine, however a non-necessary `traefik/stage/acme/storageFile` entry is created with no value.

```shell
docker run \
 -v $PWD/traefik-ssl-acme-seed.toml:/etc/traefik/traefik.toml  \
 -e "CONSUL_HTTP_TOKEN=[consul-token]" \
 traefik \
 storeconfig -c /etc/traefik/traefik.toml \
 --consul \
 --consul.endpoint=https://my.consul.node:8501 \
 --consul.prefix="traefik/stage"
```

*traefik-ssl-acme-seed.toml*
```toml
logLevel="DEBUG"
defaultEntryPoints = ["http","https"]

debug=true
insecureskipverify=true

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[web]
  address = ":8080"

[web.statistics]

[retry] # Default: (number servers in backend) -1

[docker]
  domain="docker.localhost"
  swarmmode=true
  watch=true
  exposedbydefault=false

[acme]
  email = "billy.letsencrypt-stage@my-test.com"
  onHostRule = true
  entryPoint = "https"
  storage = "traefik/stage/acme/acme.json" # the Consul KV path where to store your certificates in the KV store
  acmelogging = true
  caServer = "https://acme-staging.api.letsencrypt.org/directory" # STAGING!!
```

2. **Launched traefik service pointing to this config**

```shell
docker service create \
--name traefik-test \
--mode replicated \
--replicas 1 \
--constraint=node.role==manager \
--mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock \
--network my-net \
--publish :80 \
--publish :8080 \
--publish :443 \
-e "CONSUL_HTTP_TOKEN=[consul-acl-token]" \
traefik:1.4.5  \
--consul \
--consul.endpoint=https://my.consul.node:8501 \
--consul.prefix="traefik/stage"
```

Traefik comes up, but fails to seed the`acme.json` key with the initial JSON database (the result of the initial ACME registration/init)

3. **If I delete the empty `storageFile` key, and restart traefik service, acme.json is properly initialized.**

### What did you expect to see?

When I initially ran `storeconfig` the acme/storageFile key should have never been created in the first place

### What did you see instead?

Described above

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Traefik 1.4.5
Consul 1.0.0

Version:      v1.4.5
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-12-06_10:16:48AM
OS/Arch:      linux/amd64

**Comment by zyzop** [comment]
It's a well-known behavior. A lot of traefik guides mention this.

**Comment by bitsofinfo** [comment]
Doesn't mean it makes sense, which is what I am pointing out. If I want to start a new traefik instance pointing to KV and have it init its ACME there, there is zero need for this `storageFile` attribute when `storage` was already specified w/ a legit path, and causing it not to work....

**Comment by ldez** [comment]
FYI `StorageFile` is deprecated:
https://github.com/containous/traefik/blob/328be161d6e8cd04c1066d5884e4bf35f396b40f/acme/acme.go#L39

And it will be removed the future.

However, could you provide some logs (debug): when you restart traefik **without** delete the empty `storageFile` key.

**Comment by ldez** [comment]
Related to #927

**Comment by bitsofinfo** [comment]
@ldez, my output is same as https://github.com/containous/traefik/issues/927#issuecomment-285945397

**Comment by traefiker** [comment]
Closed by #2598.
